### ISSUE TITLE
Failure when calling `transpose` in `Prism::Translation::Parser`

### ISSUE THREAD

**Issue by andyw8** [ruby/prism#2688]
https://github.com/ruby/prism/pull/2602 seems to have introduced a regression. Given a file containing:

```
"a
b\
c"
```

running RuboCop with `ParserEngine: parser_prism` fails with:

```
element size differs (2 should be 3)
/home/spin/.bundle/shopify/ruby-3.3.0/bundler/gems/prism-3ae54cb4dbb3/lib/prism/translation/parser/compiler.rb:1521:in `transpose'
/home/spin/.bundle/shopify/ruby-3.3.0/bundler/gems/prism-3ae54cb4dbb3/lib/prism/translation/parser/compiler.rb:1521:in `visit_string_node'
```

**Comment by andyw8** [comment]
FYI @koic

**Comment by andyw8** [comment]
(I bsected to identify this commit. It works with the prior commit, 29b05579a8a966964fcd351ec630b38351ec0ef6).

**Comment by andyw8** [comment]
This seems to be resolved in latest Prism (v0.27.0).
